app/services/update_submission_service.rb

```diff
@@ -1,5 +1,9 @@
 class UpdateSubmissionService
   def self.admin_update_submission(submission, current_remote_user, params)
+    if current_partner.graduate? && params[:federal_funding_details_attributes].present?
+      @funding_details = submission.federal_funding_details
+      @funding_details.update! params[:federal_funding_details_attributes]
```

Comment:
And here again.  Does `submission.update! params` not do the nested update to `federal_funding_details`?

Reply:
Ok, you're completely right...these explicit calls to @funding_details is no longer needed. This also led me to discover other areas where some of the other things I had written were no longer necessary once we started accepting nested attributes, especially in the world of error messaging.

Sorry about the duplication! I'll keep looking for more, but please let me know if you see other examples of me reinventing the wheels rather than letting Rails do its thing.

Reply:
No problem.  I sort of forgot all the capabilities of nested attributes myself.  Here's some more documentation on the subject: https://api.rubyonrails.org/v3.2/classes/ActiveRecord/NestedAttributes/ClassMethods.html .  And yes, as you mention, this should be able to handle errors for us as well.